packages/eth-providers/src/base-provider.ts

```diff
@@ -288,12 +291,25 @@ export abstract class BaseProvider extends AbstractProvider {
   readonly finalizedBlockHashes: MaxSizeSet;
 
   network?: Network | Promise<Network>;
-  latestBlockHash: string;
-  latestFinalizedBlockHash: string;
-  latestBlockNumber: number;
-  latestFinalizedBlockNumber: number;
-  runtimeVersion: number | undefined;
-  subscriptionStarted: boolean;
+
+  #subscription: Promise<() => void> | undefined;
+  head$: Observable<Header>;
+  finalizedHead$: Observable<Header>;
+
+  readonly queue = new AsyncScheduler(AsyncAction);
+
+  get latestBlockHash(): Promise<string> {
+    return firstValueFrom(this.head$.pipe(map(header => header.hash.toHex())));
```

Comment:
just wanna double check: does `head$` emit current block header immediately on subscription? Otherwise this will return next block hash instead.

Also, I am wondering if this would degrade performance? Previously we cache these 4 variables locally, since they are used very often by all methods, and instead of querying `await api.rpc.chain.getBlockHash()` each time, we hope to achieve an instant query time. 

(correct me if I am wrong) In the Rxjs version, every query to `latestBlockHash` needs to setup a subscription, transform that to promise, resolve it, and close subscription. We basically changed a sync local variable read to an asynchronous fetch, similar to using `await api.rpc.chain.getBlockHash()`, I am not too sure what's the benefits for this?